models/HubSpotHttpError.ts

```diff
@@ -0,0 +1,46 @@
+import { AxiosResponse, InternalAxiosRequestConfig, isAxiosError } from 'axios';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export class HubSpotHttpError<T = any, D = any> extends Error {
+  public status?: number;
+  public response?: AxiosResponse<T, D>;
+  public config?: InternalAxiosRequestConfig<D>;
+  public code?: string;
+  public request?: unknown;
+  public statusText?: string;
+  public data?: T;
+
+  // Add this in so any pre-existing checks for error.isAxiosError continue to
+  // function until they can be ported over to the new error check
+  public isAxiosError = true;
+
+  constructor(message?: string, options?: ErrorOptions) {
+    super(message, options);
+    this.name = 'HubSpotHttpError';
+
+    if (options && isAxiosError(options.cause)) {
+      // Add these for backwards compatibility until we have updated all the checks
+      // in the CLI for the Axios implementation details, and then we can remove these or
+      // keep whatever we find useful in custom fields on this object
+      const { response, request, config, code } = options.cause;
+      this.response = response;
+      this.request = request;
+      this.config = config;
+
+      // Add any custom fields we feel are necessary or make our collective lives easier
+      this.code = code;
+      if (response) {
```

Comment:
For debugging purposes should we include more of the response information here? Things like headers and config maybe ([response schema](https://axios-http.com/docs/res_schema))?

Reply:
Good call, I'll add headers to the top level